crypto-jni/src/lib.rs

```diff
@@ -408,3 +467,199 @@ pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_generateCo
 
     return CryptoJniResult::Ok(payload).to_jstring(&env);
 }
+
+type SerializeParamResult = CryptoJniResult<String>;
+
+#[no_mangle]
+#[allow(non_snake_case)]
+/**
+ * The JNI wrapper for serializing receive parameters.
+ * Constructs a SerializeParamResult containg the hex encoded serialized parameter. 
+ */
+pub extern "system" fn Java_com_concordium_sdk_crypto_CryptoJniNative_serializeReceiveParameter(
+    env: JNIEnv,
+    _: JClass,
+    parameter: JString,
+    contractName: JString,
+    methodName: JString,
+    schemaBytes: jbyteArray,
+    schemaVersion: jint,
+    verboseErrors: jboolean,
+) -> jstring {
+
+    let schema = match env.convert_byte_array(schemaBytes) {
+        Ok(x) => x,
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+
+    let parameter: String = match env.get_string(parameter) {
+        Ok(java_str) => match java_str.to_str() {
+            Ok(rust_str) => String::from(rust_str),
+            Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+        },
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+    let contractName: String = match env.get_string(contractName) {
+        Ok(java_str) => match java_str.to_str() {
+            Ok(rust_str) => String::from(rust_str),
+            Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+        },
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+    let methodName: String = match env.get_string(methodName) {
+        Ok(java_str) => match java_str.to_str() {
+            Ok(rust_str) => String::from(rust_str),
+            Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+        },
+        Err(err) => return SerializeParamResult::from(err).to_jstring(&env),
+    };
+
+    let version: Option<u8> = match schemaVersion {
+        0 => Some(0),
+        1 => Some(1),
+        2 => Some(2),
+        3 => Some(3),
+        _ => None,
```

Comment:
Is this sensible? What if the provided schema version was 4?

Reply:
Ive refactored `CryptoJNINative` so the public method for serializing parameters recieves a `SchemaParameter`, `ReceiveName`/`InitName` and a `SchemaVersion`. This ensures the actual rust call is never made with illegal values e.g. 4.